### ISSUE TITLE
Add support for Transformer Optimization Scheduler

### ISSUE THREAD

**Issue by ahatamiz** [Project-MONAI/MONAI#2537]
**Is your feature request related to a problem? Please describe.**
Add schedulers for training transformer-based models (e.g. ViT). These schedulers include:

- WarmupConstantSchedule
- WarmupLinearSchedule
- WarmupCosineSchedule
- WarmupCosineWithHardRestartsSchedule

**Describe the solution you'd like**
Currently, Pytorch does not have implementations for these schedulers. However, HuggingFace library provides such implementations:
https://huggingface.co/transformers/v1.2.0/_modules/pytorch_transformers/optimization.html

**Comment by ahatamiz** [comment]
@wyli @Nic-Ma

**Comment by ahatamiz** [comment]
In particular [WarmupCosineSchedule](https://huggingface.co/transformers/v1.2.0/_modules/pytorch_transformers/optimization.html#WarmupCosineSchedule) is widely used for scheduling the optimization in training of Transformers. This scheduler applies linear warmup and then cosine decay. The closest Pytorch implementation is [CosineAnnealingWarmRestarts](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html#CosineAnnealingWarmRestarts) which is quite different as it does not apply the linear warmup.
